File: Inverted-Index/src/keywords/Union.java
package keywords;
import java.util.Set;

import file_handler.Doc;

public class Union extends Keyword {

    public Union(String word){
        super(word);
    }

    @Override
    public Set<Doc> operate(Set<Doc> docs, Set<Doc> newDocs) {
        if(docs == null) {
            return newDocs;
        }
        if(newDocs != null)
            docs.addAll(newDocs);
        return docs;
    }
}

File: Phase5-TDD/InvertedIndex/Keywords/Union.cs
using System.Globalization;
using System;
using System.Collections.Generic;
using InvertedIndex.FileHandler;

namespace InvertedIndex.Keywords
{
    public class Union : Keyword
    {
        private string word;

        public Union(string word)
        {
            this.word = word;
        }

        public virtual string GetWord()
        {
            return word;
        }
        public virtual HashSet<Doc> Operate(HashSet<Doc> docs, HashSet<Doc> newDocs)
        {
            if(docs == null) {
            return newDocs;
            }
            if(newDocs != null)
                docs.UnionWith(newDocs);
            return docs;
        }
    }
}